awx_collection/plugins/modules/tower_application.py

```diff
@@ -0,0 +1,170 @@
+
+
+DOCUMENTATION = '''
+---
+module: tower_application
+author: "Geoffrey Bacheot (@jffz)"
+short_description: create, update, or destroy Ansible Tower applications
+description:
+    - Create, update, or destroy Ansible Tower applications. See
+      U(https://www.ansible.com/tower) for an overview.
+options:
+    name:
+      description:
+        - Name of the application.
+      required: True
+      type: str
+    description:
+      description:
+        - Description of the application.
+      type: str
+    authorization_grant_type:
+      description:
+        - The grant type the user must use for acquire tokens for this application.
+      default: "password"
+      choices: ["password", "authorization-code"]
+      type: str
+      required: True
+    client_type:
+      description:
+        - Set to public or confidential depending on how secure the client device is.
+      choices: ["public", "confidential"]
+      type: str
+      required: True
+    organization:
+      description:
+        - Name of organization for application.
+      type: str
+      required: True
+    redirect_uris:
+      description:
+        - Allowed urls list, space separated. Required when authorization-grant-type=authorization-code
+      default: "present"
+      choices: ["present", "absent"]
+      type: str
+    state:
+      description:
+        - Desired state of the resource.
+      default: "present"
+      choices: ["present", "absent"]
+      type: str
+    skip_authorization:
+      description:
+        - Set True to skip authorization step for completely trusted applications.
+      default: false
+      type: bool
+    username:
+      description:
+        - Username for this credential. ``access_key`` for AWS.
+        - Deprecated, please use inputs
+      type: str
+'''
+
+
+EXAMPLES = '''
+- name: Add Foo application
+  tower_application:
+    name: "Foo"
+    description: "Foo bar application"
+    organization: "test"
+    state: present
+    authorization-grant-type: password
+    client-type: public
+
+- name: Add Foo application
+  tower_application:
+    name: "Foo"
+    description: "Foo bar application"
+    organization: "test"
+    state: present
+    authorization-grant-type: authorization-code
+    client-type: confidential
+    redirect_uris: http://tower.com/api/v2/
+'''
+
+import time
+
+from ..module_utils.tower_api import TowerModule
+
+
+def main():
+    # Any additional arguments that are not fields of the item can be added here
+    argument_spec = dict(
+        name=dict(required=True),
+        description=dict(),
+        authorization_grant_type=dict(required=True),
+        client_type=dict(required=True, choices=['public', 'confidential']),
+        organization=dict(required=True),
+        redirect_uris=dict(type="list", elements='str'),
+        state=dict(choices=['present', 'absent'], default='present'),
+        skip_authorization=dict(type=bool),
+        username=dict()
```

Comment:
It looks like you are never using `username` so we should just be able to remove it.

Reply:
deleted in latest commit